### DISCUSSION TITLE
Request for Assistance with Pipeflow Example and RBC Dimension Handling

### DISCUSSION THREAD

**Discussion by rishabhbhatt233** [UvaCsl/HemoCell#124] [General discussion]
I have an STL file of a pipe (pipe.stl) for the "pipeflow" example, and I want to add red blood cells (RBCs) and flow to the simulation. Can I generate the rbc.pos file using an in-built script?

Additionally, when I load the STL file for the RBC, it becomes slightly larger than the RBC generated using the "MESH_FROM_SPHERE" method. How can I handle the dimensions of RBC.stl and pipe.stl after loading to ensure they are correctly sized?

**Comment by mojtabaAmiraslanpour** [comment]
> I have an STL file of a pipe (pipe.stl) for the "pipeflow" example, and I want to add red blood cells (RBCs) and flow to the simulation. Can I generate the rbc.pos file using an in-built script?
> 
> Additionally, when I load the STL file for the RBC, it becomes slightly larger than the RBC generated using the "MESH_FROM_SPHERE" method. How can I handle the dimensions of RBC.stl and pipe.stl after loading to ensure they are correctly sized?

To produce cell .pos files, you can use the packCells tool. You can find it [here](https://github.com/UvaCsl/HemoCell/blob/master/tools/packCells/packCells.cpp).

When you import your own STL file for RBC, you can set the size with the radius of the cell tag in the same file (RBC.xml).

**Reply by rishabhbhatt233** [reply]
Thank you
How can I make cell.pos file for stl files like[link](https://github.com/UvaCsl/HemoCell/blob/d91698721f7a06166cb6ac8d22fa2f2f4baa4ed7/cases/stl_preinlet/tube.stl), because packCells needs domain size, cause I am instated to simulate in artery.
For the case of pipeflow I make cell.pos file I have 25 number of RBCs but it loads only 10 RBCs with in-build "RBC_FROM_SPHERE" module, so if I make .pos using Ht% will it load correct RBCs for that.

Also how can I get the output of that tube or artery for visualization?

**Reply by mojtabaAmiraslanpour** [reply]
You can use a higher Ht so that more RBCs remain in the domain. Some cells will be deleted simply because they are on the boundary. You can also manually find the locations and angles of the RBCs and put them in the RBC.pos file if you only have 25 RBCs.
